core/cap-0023.md

```diff
@@ -239,146 +293,248 @@ default:
+   not have at least `min(1, condition.size()-1) * baseReserve` available
+   balance of native asset (see CAP-0003)
+2. Deduct `min(1, condition.size()-1) * baseReserve` of native asset from
+   `sourceAccount`
+3. Fail with `CREATE_CLAIMABLE_BALANCE_NO_TRUST` if the `sourceAccount` does not
+   have a trust line for `asset`
+4. Fail with `CREATE_CLAIMABLE_BALANCE_NOT_AUTHORIZED` if the `sourceAccount` is
+   not authorized for `asset`
+5. Fail with `CREATE_CLAIMABLE_BALANCE_UNDERFUNDED` if the `sourceAccount` does
+   not have at least `amount` available balance of `asset`
+6. Deduct `amount` of `asset` from `sourceAccount`
+7. Create a balance entry with the following properties:
+    - `balanceID` of type `CLAIMABLE_BALANCE_ID_TYPE_V0`. `balanceID.v0()` equal
+       to the SHA256 hash of the concatenation of the `sourceAccount` of the
+       transaction, the `seqNum` of the transaction, and the index of this
+       operation in the transaction
+    - `condition` as specified in the operation, with the exception that
+        - `CLAIM_PREDICATE_BEFORE_RELATIVE_TIME` will be converted to
+          `CLAIM_PREDICATE_BEFORE_ABSOLUTE_TIME` by adding `relBefore` to the
+          `closeTime` in the `LedgerHeader`
+        - `CLAIM_PREDICATE_BEFORE_RELATIVE_TIME` will be converted to
+          `CLAIM_PREDICATE_AFTER_ABSOLUTE_TIME` by adding `relAfter` to the
+          `closeTime` in the `LedgerHeader`
+    - `threshold` as specified in the operation
+    - `reserve` equal to `min(1, condition.size()-1) * baseReserve`
+    - `asset` as specified in the operation
+    - `amount` as specified in the operation
+    - `authorize` equal to 0
+8. Succeed with `CREATE_CLAIMABLE_BALANCE_SUCCESS`
+
+`CreateClaimableBalanceOp` requires medium threshold because it can be used to
+send funds.
+
+### ClaimClaimableBalanceOp
+A `ClaimableBalanceEntry` can only be deleted by the `ClaimClaimableBalanceOp`
+operation. `ClaimClaimableBalanceOp` cannot be malformed. The behavior of
+`ClaimClaimableBalanceOp` is as follows:
+
+1. Fail with `CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST` if there is no
+   `ClaimableBalanceEntry` matching `balanceID`.
+2. Calculate the total weight of `condition` with `destination = sourceAccount`
+   and satisfied predicates.
+3. Fail with `CLAIM_CLAIMABLE_BALANCE_CANNOT_CLAIM` if the total weight is less
+   than `threshold`.
+4. Fail with `CLAIM_CLAIMABLE_BALANCE_LINE_FULL` if the `sourceAccount` does not
+   have at least `reserve` available limit of native asset (see CAP-0003)
+5. Add `reserve` of native asset to the `sourceAccount`
+6. Skip to step 14 if `asset` is native
+7. Skip to step 16 if `sourceAccount` is the issuer for `asset`
+8. Skip to step 12 if `sourceAccount` has a trust line for `asset`
+9. Fail with `opTOO_MANY_SUBENTRIES` if `sourceAccount` is already at the
+   subentry limit
+10. Fail with `CLAIM_CLAIMABLE_BALANCE_LOW_RESERVE` if the `sourceAccount` does
```

Comment:
If we fail at this point is the affect of step 5 something that has still taken place, or are all of these steps atomic?